specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2022-01-01-preview/fluxconfiguration.json

```diff
@@ -0,0 +1,1096 @@
+          "type": "string",
+          "x-nullable": true
+        },
+        "httpsCACert": {
+          "description": "Base64-encoded HTTPS certificate authority contents used to access git private git repositories over HTTPS",
+          "type": "string",
+          "x-nullable": true
+        },
+        "localAuthRef": {
+          "description": "Name of a local secret on the Kubernetes cluster to use as the authentication secret rather than the managed or user-provided configuration secrets.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "RepositoryRefDefinition": {
+      "description": "The source reference for the GitRepository object.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "branch": {
+          "description": "The git repository branch name to checkout.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "tag": {
+          "description": "The git repository tag name to checkout. This takes precedence over branch.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "semver": {
+          "description": "The semver range used to match against git repository tags. This takes precedence over tag.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "commit": {
+          "description": "The commit SHA to checkout. This value must be combined with the branch name to be valid. This takes precedence over semver.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "BucketDefinition": {
+      "description": "Parameters to reconcile to the GitRepository source kind type.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "url": {
+          "description": "The URL to sync for the flux configuration S3 bucket.",
+          "type": "string"
+        },
+        "bucketName": {
+          "description": "The bucket name to sync from the url endpoint for the flux configuration.",
+          "type": "string"
+        },
+        "insecure": {
+          "description": "Specify whether to use insecure communication when puling data from the S3 bucket.",
+          "type": "boolean",
+          "default": true
+        },
+        "timeoutInSeconds": {
+          "description": "The maximum time to attempt to reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "syncIntervalInSeconds": {
+          "description": "The interval at which to re-reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "accessKey": {
```

Comment:
mark this as a secret appropriately assuming it is one (and ensure you don't return it in the response or any future GETs):  [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi013-secrets-must-be-annotated-with-x-ms-secret-and-x-ms-mutability](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi013-secrets-must-be-annotated-with-x-ms-secret-and-x-ms-mutability)

Reply:
This is not a secret, this is similar to username for an S3 bucket